onscan.js

```diff
@@ -438,7 +603,7 @@
 			}
 			
 			if(bScanFinished){
-				onScan._validateScanCode(this, oVars.accumulatedString);
+				await onScan._validateScanCode(this, oVars.accumulatedString);
 				oVars.testTimer=false;
```

Comment:
@bashar-odeh shouldn't this be cleared?
```suggestion
				clearTimeout(oVars.testTimer);
```

Reply:
it's being cleared in the code right above it 

```javascript
			if(oVars.testTimer){ 
				clearTimeout(oVars.testTimer);
			}
```

Reply:
oh ok, but do we need `oVars.testTimer=false;`?

Reply:
it's being reset to false which is the default value for it, also it should be something that is false since it can be used in other places if it was still the old ID and not cleared. such as in our case, if it was not cleared it would try to clear it again.